frontend/src/metabase/static-viz/components/XYChart/utils/values.ts

```diff
@@ -0,0 +1,33 @@
+import { getY } from "./series";
+
+import type { TextProps } from "@visx/text";
+import type { HydratedSeries } from "../types";
+import { measureText } from "metabase/static-viz/lib/text";
+
+// Use the same logic as in https://github.com/metabase/metabase/blob/72d23d2b8e5a1f93b288e5a8738758c9b05d3766/frontend/src/metabase/visualizations/lib/chart_values.js#L318
+export function getValueStep(
+  series: HydratedSeries[],
+  valueFormatter: (value: number) => string,
+  valueProps: Partial<TextProps>,
+  chartWidth: number,
+) {
+  if (series.length === 0) {
+    return 1;
+  }
+  const maxSeriesLength = Math.max(...series.map(serie => serie.data.length));
+  const LABEL_PADDING = 6;
+  const MAX_SAMPLE_SIZE = 30;
+  const sampleStep = Math.ceil(maxSeriesLength / MAX_SAMPLE_SIZE);
+  const sample = series[0].data.filter((_, index) => index % sampleStep === 0);
+  const totalWidth = sample.reduce(
+    (sum, sampledData) =>
+      sum +
+      measureText(
```

Comment:
Use `measureText` to calculate the width instead of using `getBoundingClientRect` I haven't tested if `getBoundingClientRect` doesn't work on BE and that's why we need `measureText` in the first place, maybe @alxnddr would know?

Reply:
No, you cannot use it since there is no actual rendering of any real DOM. That would make things a lot easier if we could use `getBoundingClientRect`

Reply:
that makes sense.